firmware/src/control.cpp

```diff
@@ -221,6 +221,11 @@ void runVelocityControl() {
   }
 }
 
+static void setScaledDuty(uint8_t phase, float duty) {
+  // Scale the duty cycle to prevent interference of gate on-time with ADC sampling
+  gate_driver.setPWMDutyCycle(phase, duty*max_duty_cycle);
```

Comment:
Does the duty cycle actually need to be limited, since we have 3 shunts?
If `duty cycle A > max_duty_cycle`, seems like we can just toss out `current A` from the ADC reading and re-compute it as `-(current B + current C)`.

Also feels a little weird to be scaling the duty cycles by a constant, since they're supposed to correspond to the `vd` and `vq` voltages.

Reply:
Ooo throwing out the ISNS reading of the highest duty cycle channel (while above a threshold) is a very cool idea... besides the loss of some redundancy I don't see a major downside, except that it might make debugging harder.